litestar/_openapi/schema_generation/schema.py

```diff
@@ -142,22 +142,22 @@
 }
 
 
-def _get_type_schema_name(value: Any) -> str:
+def _get_type_schema_name(field_definition: FieldDefinition) -> str:
     """Extract the schema name from a data container.
 
     Args:
-        value: A data container
+        field_definition: A field definition instance.
 
     Returns:
         A string
     """
 
-    if name := getattr(value, "__schema_name__", None):
+    if name := getattr(field_definition.annotation, "__schema_name__", None):
```

Comment:
Would this have to be `field_definition.origin or field_definition.annotation` since `Foo[T]` might not have `__schema_name__`?

Reply:
Well I don't think this is any more broken than it was, is it? B/c in every case that we called the function before this change we were passing `field_definition.annotation` through.

However, if you mean this:

```python-console
>>> from typing import Generic
>>> from typing import TypeVar
>>> T = TypeVar("T")
>>> class GenType(Generic[T]):
...     __schema_name__ = "abc"
... 
>>> GenType[int].__schema_name__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/peter/.pyenv/versions/3.8.17/lib/python3.8/typing.py", line 760, in __getattr__
    raise AttributeError(attr)
AttributeError: __schema_name__
```
Then, yeah - I think you are right.